### PR TITLE
Fix - percent encode ' ', '%', '<', '>', '[', ']', '#', '{', '}', '^', '`', '|' by default in the fragment

### PR DESCRIPTION
**Pull request by takayahilton**
fragment    = *( pchar / "/" / "?" )

### CODE REVIEW

File: shared/src/main/scala/com/netaporter/uri/encoding/PercentEncoder.scala [hunk 1]
```diff
@@ -23,19 +23,20 @@ case class PercentEncoder(charsToEncode: Set[Char] = DEFAULT_CHARS_TO_ENCODE) ex
...
   )
 
   val PATH_CHARS_TO_ENCODE = Set (
-    ' ', '%', '<', '>', '[', ']', '#', '%', '{', '}', '^', '`', '|', '?'
+    ' ', '%', '<', '>', '[', ']', '#', '{', '}', '^', '`', '|', '?'
   )
 
   val QUERY_CHARS_TO_ENCODE = Set (
-    ' ', '%', '<', '>', '[', ']', '#', '%', '{', '}', '^', '`', '|', '&', '\\', '+', '='
+    ' ', '%', '<', '>', '[', ']', '#', '{', '}', '^', '`', '|', '&', '\\', '+', '='
```
Comment (takayahilton) [inline shared/src/main/scala/com/netaporter/uri/encoding/PercentEncoder.scala]: duplicated %

File: shared/src/test/scala/com/netaporter/uri/EncodingTests.scala [hunk 2]
```diff
@@ -71,8 +71,8 @@ class EncodingTests extends FlatSpec with Matchers {
   }
 
   "Fragments" should "be percent encoded" in {
-    val uri = "http://theon.github.com/uris-in-scala.html" ? ("chinese" -> "网址")
-    uri.toString should equal ("http://theon.github.com/uris-in-scala.html?chinese=%E7%BD%91%E5%9D%80")
+    val uri = "http://theon.github.com/uris-in-scala.html" `#` "chinese# 网址"
```
Comment (takayahilton) [inline shared/src/test/scala/com/netaporter/uri/EncodingTests.scala]: from ? to `#`

Comment (theon) [inline shared/src/test/scala/com/netaporter/uri/EncodingTests.scala]: Good catch!

**Comment by theon** [comment]
Thanks for the contribution!

**Comment by theon** [comment]
Released as version `0.5.7`

**Pull request merged**